dev/run-tests.py

```diff
@@ -610,8 +610,14 @@ def main():
 
     # Install SparkR
     should_only_test_modules = opts.modules is not None
-    if not should_only_test_modules:
+    test_modules = []
+    if should_only_test_modules:
+        str_test_modules = [m.strip() for m in opts.modules.split(",")]
+        test_modules = [m for m in modules.all_modules if m.name in str_test_modules]
+
+    if not should_only_test_modules or modules.sparkr in test_modules:
```

Comment:
I piggy-back this change together. We don't need to install SparkR when `--modules` is specified except that it includes `sparkr`.